File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int i=0;
        while(i<nums.size() && nums[i]==0) i+=1;
        int j=i+1;
        //First bring 0 together
        while(j<nums.size()) {
             while( i<nums.size() && nums[i]==0) {
                    i++;
                    j++;
                }
                if(j>=nums.size()) break;
            if(nums[j]==0) {
                swap(nums[i],nums[j]);
                i+=1;
                j+=1;
               
            } else {
                j+=1;
            }
        }
        
        j=i+1;
        while(j<nums.size()) {
             while( i<nums.size() && nums[i]==1) {
                    i++;
                    j++;
                }
                if(j>=nums.size()) break;
            if(nums[j]==1) {
                swap(nums[i],nums[j]);
                i+=1;
                j+=1;
               
            } else {
                j+=1;
            }
        }
       
    }
};

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        n = len(nums)
        for j in range(n):
            while i<j and nums[i]==0: i+=1
            if nums[j]==0:
                nums[i],nums[j] = nums[j],nums[i]
                i+=1
        
        i=0
        j = 0
        for j in range(n):
            while i<j and (nums[i]==0 or nums[i]==1): i+=1
            if nums[j] ==1:
                nums[i],nums[j] = nums[j],nums[i]
        return nums

        